pkgs/build-support/nix-gitignore/default.nix

```diff
@@ -147,22 +147,27 @@ in rec {
         ' sh {} \; > $out
       '');
 
-  withGitignoreFile = patterns: root:
-    lib.toList patterns ++ [(root + "/.gitignore")];
+  withGitIgnoredFilesAndDirs = patterns: root:
+    lib.toList patterns ++ [(root + "/.gitignore") (".git")];
 
-  withRecursiveGitignoreFile = patterns: root:
+  withRecursiveGitIgnoredFilesAndDirs = patterns: root:
```

Comment:
Are these renames necessary? To me, it feels like that would break compatibility with existing code for no good reason.

But since you already renamed them in the repo, maybe it would be good to rename for consistency. Perhaps we could at least add an alias for some time, with a warning (using `lib.warn`) to make transitioning easier?